packages/editor/src/utils/media-upload/media-upload.js

```diff
@@ -137,17 +137,12 @@ export function mediaUpload( {
 		return createMediaFromFile( mediaFile, additionalData )
 			.then( ( savedMedia ) => {
 				const mediaObject = {
+					...omit( savedMedia, [ 'alt_text', 'source_url' ] ),
 					alt: savedMedia.alt_text,
 					caption: get( savedMedia, [ 'caption', 'raw' ], '' ),
-					id: savedMedia.id,
-					link: savedMedia.link,
 					title: savedMedia.title.raw,
 					url: savedMedia.source_url,
-					mediaDetails: {},
 				};
```

Comment:
Please note that there is this subtle difference between `media_details` and `mediaDetails` in both objects. In addition, we no longer normalize `mediaDetails.sizes`

Reply:
Nice catch @gziolo. We are not using mediaDetails anywhere in our code but a plugin may be using it.
I feel we should use what is returned by the rest API, so we are consistent with media_details used in other rest API responses. I added a deprecation so we can keep the previous version for some versions.
Thank you for your review!

Reply:
Can we auto-map keys instead? We should really stop adding deprecations at this stage.